contracts/Comptroller/Diamond/facets/PolicyFacet.sol

```diff
@@ -249,27 +276,51 @@ contract PolicyFacet is IPolicyFacet, XVSRewardsHelper {
 
         if (isForcedLiquidationEnabled[vTokenBorrowed] || isForcedLiquidationEnabledForUser[borrower][vTokenBorrowed]) {
             if (repayAmount > borrowBalance) {
-                return uint(Error.TOO_MUCH_REPAY);
+                revert TooMuchRepay();
             }
-            return uint(Error.NO_ERROR);
+            return NO_ERROR;
         }
 
         /* The borrower must have shortfall in order to be liquidatable */
-        (Error err, , uint256 shortfall) = getHypotheticalAccountLiquidityInternal(borrower, VToken(address(0)), 0, 0);
-        if (err != Error.NO_ERROR) {
-            return uint256(err);
-        }
+        (, , uint256 shortfall) = getHypotheticalAccountLiquidityInternal(
+            borrower,
+            VToken(address(0)),
+            0,
+            0,
+            this.getLiquidationThreshold
+        );
+
+        (, uint256 liquidationThresholdAvg, uint256 totalCollateral, ) = getHypotheticalHealthSnapshot(
```

Comment:
I think that if `getHypotheticalHealthSnapshot` returns the `shortfall` amount, we could avoid the previous call to `getHypotheticalAccountLiquidityInternal`, saving a significant amount of gas

Reply:
[56c25c2](https://github.com/VenusProtocol/venus-protocol/pull/604/commits/56c25c2c17ad2d3705d4b4bde4f70ef98352a37c)